models/post_repo.go

```diff
@@ -0,0 +1,83 @@
+package models
+
+import "fmt"
+
+// PostRepository handles the CRUD for post.
+type PostRepository interface {
+	Create(*Post) error
+	Get(*Query) (*Post, error)
+	GetAll(*Query) (*[]Post, error)
+	Update(*Post) error
+}
+
+// DBPostRepository ...
+type DBPostRepository struct{}
+
+// NewDBPostRepository ...
+func NewDBPostRepository() PostRepository {
+	return DBPostRepository{}
+}
+
+// Create handles the creation of post.
+func (repo DBPostRepository) Create(post *Post) error {
+	tx := DB.MustBegin()
+	_, err := tx.NamedExec("INSERT INTO post (body, caption, user_id, ctime, utime) VALUES (:body, :caption, :user_id, :ctime, :utime)", post)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Get handles the fetching of a single post record.
+func (repo DBPostRepository) Get(query *Query) (*Post, error) {
+	post := Post{}
+
+	queryString := "SELECT * FROM post WHERE"
+
+	if query.ByID != nil {
+		queryString = queryString + " " + fmt.Sprintf("id = %d", query.ByID)
+	}
+
+	if query.ByColumn != nil {
+		queryString = queryString + " " + fmt.Sprintf("%v = %v", query.ByColumn.Column, query.ByColumn.Value)
+	}
+
+	err := DB.Get(&post, queryString)
+	if err != nil {
+		return nil, err
+	}
+
+	return &post, nil
+}
+
+// GetAll handles the fetching ofpost records.
+func (repo DBPostRepository) GetAll(query *Query) (*[]Post, error) {
+	posts := []Post{}
+
+	queryString := "SELECT * FROM post WHERE"
```

Comment:
We might run into an SQL Syntax issue when nil is passed for query, the `queryString` might not be appended. Maybe `SELECT * FROM post WHERE` will throw an error? Is it possible to just use `ByID` for the mean time?

Change line 35 to `queryString := "SELECT * FROM post"`
and then change line 38 to `queryString = queryString + " " + fmt.Sprintf("WHERE id = %d", query.ByID)`
and then just remove the entire code block lines 41-43?

Reply:
What do you think @rbo13 ?

Reply:
sure I can do that thanks for feedback

Reply:
@kenkoii how about we just stick to **ByColumn** since we can query **ID** too. We just remove the **ByID** clause. What do you think?